server/service/transport_error.go

```diff
@@ -108,54 +117,70 @@ func encodeError(ctx context.Context, err error, w http.ResponseWriter) {
 		}
 
 		enc.Encode(errMap)
-		return
-	}
-
-	type notFoundError interface {
-		error
-		IsNotFound() bool
-	}
-	if e, ok := err.(notFoundError); ok {
+	case notFoundErrorInterface:
 		je := jsonError{
 			Message: "Resource Not Found",
 			Errors:  baseError(e.Error()),
 		}
 		w.WriteHeader(http.StatusNotFound)
 		enc.Encode(je)
-		return
-	}
-
-	type existsError interface {
-		error
-		IsExists() bool
-	}
-	if e, ok := err.(existsError); ok {
+	case existsErrorInterface:
 		je := jsonError{
 			Message: "Resource Already Exists",
 			Errors:  baseError(e.Error()),
 		}
 		w.WriteHeader(http.StatusConflict)
 		enc.Encode(je)
-		return
-	}
+	case *mysql.MySQLError:
+		je := jsonError{
+			Message: "Validation Failed",
+			Errors:  baseError(e.Error()),
+		}
+		w.WriteHeader(http.StatusUnprocessableEntity)
+		enc.Encode(je)
```

Comment:
What do you think of making duplicate key errors return `409`, foreign key errors return `422`, and all other MySQL errors return `500`?